### PR TITLE
Document zlib install

### PR DESCRIPTION
**Pull request by mnapoli**
Follow-up for #110

### CODE REVIEW

File: php-83/Dockerfile [hunk 1]
```diff
@@ -72,26 +72,18 @@ RUN mkdir -p ${BUILD_DIR}  \
 
 ###############################################################################
 # ZLIB Build
+# We compile a newer version because Lambda uses an old version (1.2.7) that
+# has a security vulnerability (CVE-2022-37434).
+# See https://github.com/brefphp/aws-lambda-layers/pull/110
+# Can be removed once Lambda updates their version.
```
Comment (GrahamCampbell) [inline php-83/Dockerfile:78]: I don't think they ever will, for BC reasons. We'll likely have to wait till Lambda gets AL 2023, if that ever happens. ;)

Comment (mnapoli) [reply php-83/Dockerfile:78]: @GrahamCampbell yep these are my notes for AL2023 😄

File: php-80/Dockerfile [hunk 2]
```diff
@@ -80,7 +80,7 @@ ENV VERSION_ZLIB=1.3
 ENV ZLIB_BUILD_DIR=${BUILD_DIR}/zlib
 RUN set -xe; \
     mkdir -p ${ZLIB_BUILD_DIR}; \
-    curl -Ls https://zlib.net/zlib-${VERSION_ZLIB}.tar.xz \
+    curl -Ls http://zlib.net/zlib-${VERSION_ZLIB}.tar.xz \
```
Comment (GrahamCampbell) [inline php-80/Dockerfile:83]: We could try `https://github.com/madler/zlib/releases/download/v${VERSION_ZLIB}/zlib-${VERSION_ZLIB}.tar.gz` instead?

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: php-80/Dockerfile [hunk 3]
```diff
@@ -71,26 +71,18 @@ RUN mkdir -p ${BUILD_DIR}  \
...
-#   - xml2
 ENV VERSION_ZLIB=1.3
-ENV ZLIB_BUILD_DIR=${BUILD_DIR}/xml2
-
+ENV ZLIB_BUILD_DIR=${BUILD_DIR}/zlib
 RUN set -xe; \
     mkdir -p ${ZLIB_BUILD_DIR}; \
-# Download and upack the source code
-    curl -Ls http://zlib.net/zlib-${VERSION_ZLIB}.tar.xz \
+    curl -Ls https://github.com/madler/zlib/releases/download/v${VERSION_ZLIB}/zlib-${VERSION_ZLIB}.tar.gz \
```
Comment (GrahamCampbell) [inline php-80/Dockerfile:83]: oh, whoops. that is .gz, but the extract command expends .xz

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]

File: php-80/Dockerfile [hunk 4]
```diff
@@ -71,35 +71,25 @@ RUN mkdir -p ${BUILD_DIR}  \
...
-
-# Configure the build
 RUN set -xe; \
     make distclean \
  && CFLAGS="" \
     CPPFLAGS="-I${INSTALL_DIR}/include  -I/usr/include" \
     LDFLAGS="-L${INSTALL_DIR}/lib64 -L${INSTALL_DIR}/lib" \
     ./configure \
-    --prefix=${INSTALL_DIR} \
-    --64
```
Comment (mnapoli) [inline php-80/Dockerfile:101]: @GrahamCampbell I removed this line (after reading https://github.com/madler/zlib/issues/143) at it seemed to do the trick. Do you see any reason to be worried?

If not I'll merge and release.

Comment (GrahamCampbell) [reply php-80/Dockerfile:101]: Ah yeh, I think i did something similar when making vapor work with arm. This is fine. 👍

**Review by GrahamCampbell — APPROVED** [review]
Thanks for finishing off my jank.

File: php-80/Dockerfile [hunk 5]
```diff
@@ -71,35 +71,25 @@ RUN mkdir -p ${BUILD_DIR}  \
...
-
+ENV ZLIB_BUILD_DIR=${BUILD_DIR}/zlib
 RUN set -xe; \
     mkdir -p ${ZLIB_BUILD_DIR}; \
-# Download and upack the source code
-    curl -Ls http://zlib.net/zlib-${VERSION_ZLIB}.tar.xz \
-  | tar xJC ${ZLIB_BUILD_DIR} --strip-components=1
-
-# Move into the unpackaged code directory
+    curl -Ls https://github.com/madler/zlib/releases/download/v${VERSION_ZLIB}/zlib-${VERSION_ZLIB}.tar.gz \
```
Comment (GrahamCampbell) [inline php-80/Dockerfile:83]: We could also just use the `.tar.xz` version of this URL, to keep the extract command the same.

**Pull request merged**